src/index.ts

```diff
@@ -123,9 +123,15 @@ export function regexp(pattern: RegExp): Parser<string> {
 	});
 }
 
-export function seq<T extends Parser<any>[]>(parsers: T): Parser<any[]>
-export function seq<T extends Parser<any>[]>(parsers: T, select: number): Parser<any>
-export function seq<T extends Parser<any>[]>(parsers: T, select?: number): Parser<any[] | any> {
+type SeqResultItem<T> = T extends Parser<infer R> ? R : never;
+type SeqResult<T> = T extends [infer Head, ...infer Tail] ? [SeqResultItem<Head>, ...SeqResult<Tail>] : [];
+export function seq<T extends Parser<any>[]>(parsers: [...T]): Parser<SeqResult<[...T]>>;
+export function seq<T extends Parser<any>[], U extends number>(parsers: [...T], select: U): T[U];
+export function seq(parsers: Parser<any>[], select?: number | undefined) {
```

Comment:
ここのundefinedは明示的に指定しなくて良いかも？
```suggestion
export function seq(parsers: Parser<any>[], select?: number) {
```

Reply:
今きづいてけした！！！！！！！！！！！！